File: SDKs/amazonaws.com/transcribe/2017-10-26/java/lib/src/main/java/openapisdk/models/shared/MedicalTranscriptionSetting.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * MedicalTranscriptionSetting
 * Optional settings for the <a>StartMedicalTranscriptionJob</a> operation.
**/
public class MedicalTranscriptionSetting {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ChannelIdentification")
    public Boolean channelIdentification;
    public MedicalTranscriptionSetting withChannelIdentification(Boolean channelIdentification) {
        this.channelIdentification = channelIdentification;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("MaxAlternatives")
    public Long maxAlternatives;
    public MedicalTranscriptionSetting withMaxAlternatives(Long maxAlternatives) {
        this.maxAlternatives = maxAlternatives;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("MaxSpeakerLabels")
    public Long maxSpeakerLabels;
    public MedicalTranscriptionSetting withMaxSpeakerLabels(Long maxSpeakerLabels) {
        this.maxSpeakerLabels = maxSpeakerLabels;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ShowAlternatives")
    public Boolean showAlternatives;
    public MedicalTranscriptionSetting withShowAlternatives(Boolean showAlternatives) {
        this.showAlternatives = showAlternatives;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ShowSpeakerLabels")
    public Boolean showSpeakerLabels;
    public MedicalTranscriptionSetting withShowSpeakerLabels(Boolean showSpeakerLabels) {
        this.showSpeakerLabels = showSpeakerLabels;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("VocabularyName")
    public String vocabularyName;
    public MedicalTranscriptionSetting withVocabularyName(String vocabularyName) {
        this.vocabularyName = vocabularyName;
        return this;
    }
}

File: SDKs/amazonaws.com/transcribe/2017-10-26/python/src/sdk/models/shared/medicaltranscriptionsetting.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class MedicalTranscriptionSetting:
    r"""MedicalTranscriptionSetting
    Optional settings for the <a>StartMedicalTranscriptionJob</a> operation.
    """
    
    channel_identification: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ChannelIdentification') }})
    max_alternatives: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('MaxAlternatives') }})
    max_speaker_labels: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('MaxSpeakerLabels') }})
    show_alternatives: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ShowAlternatives') }})
    show_speaker_labels: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ShowSpeakerLabels') }})
    vocabulary_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('VocabularyName') }})
    
